File: app/src/main/java/com/home/young/myPassword/application/IOHelper.java
package com.home.young.myPassword.application;

import org.apache.http.util.EncodingUtils;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;

/**
 * Created by YOUNG on 16/5/12.
 */
public class IOHelper {

    public static String readSDFile(String fileName) throws IOException {

        File file = new File(fileName);

        FileInputStream fis = new FileInputStream(file);

        int length = fis.available();

        byte [] buffer = new byte[length];
        fis.read(buffer);

        String res = EncodingUtils.getString(buffer, "UTF-8");

        fis.close();
        return res;
    }


    public static void writeSDFile(String fileName, String write_str) throws IOException{

        File file = new File(fileName);

        FileOutputStream fos = new FileOutputStream(file);

        byte [] bytes = write_str.getBytes();

        fos.write(bytes);

        fos.close();
    }

    public static  void deleteSDFile(String fileName) throws IOException{
        File file = new File(fileName);
        file.delete();
    }
}

File: app/src/main/kotlin/org/tianjyan/pwd/application/IOHelper.kt
package org.tianjyan.pwd.application

import org.apache.http.util.EncodingUtils
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException

object IOHelper {

    @Throws(IOException::class)
    fun readSDFile(fileName: String): String {

        val file = File(fileName)

        val fis = FileInputStream(file)

        val length = fis.available()

        val buffer = ByteArray(length)
        fis.read(buffer)

        val res = EncodingUtils.getString(buffer, "UTF-8")

        fis.close()
        return res
    }


    @Throws(IOException::class)
    fun writeSDFile(fileName: String, writeStr: String) {

        val file = File(fileName)

        val fos = FileOutputStream(file)

        val bytes = writeStr.toByteArray()

        fos.write(bytes)

        fos.close()
    }

    fun deleteSDFile(fileName: String) {
        val file = File(fileName)
        file.delete()
    }
}